src/Symfony/Component/Routing/Tests/Matcher/UrlMatcherTest.php

```diff
@@ -941,6 +941,20 @@ public function testRestrictiveTrailingRequirementWithStaticRouteAfter()
         $this->assertEquals(['_route' => 'a', '_' => '/'], $matcher->match('/hello/'));
     }
 
+    public function testUtf8PatternMatchAndParameterReturn()
+    {
+        $collection = new RouteCollection();
+        $collection->add('foo', new Route('/foo/{bär}', [], [], ['utf8' => true]));
+        $matcher = $this->getUrlMatcher($collection);
+        try {
+            $matcher->match('/no-match');
+            $this->fail();
+        } catch (ResourceNotFoundException $e) {
+        }
+
+        $this->assertEquals(['_route' => 'foo', 'bär' => 'baz'], $matcher->match('/foo/baz'));
```

Comment:
assertSame whenever possible

Reply:
I'm not sure that it's possible to use assertSame here. $matcher->match() doesn't guarantee the order of elements in the array. And assertEquals does it.